### ISSUE TITLE
Specify custom PID at patient creation

### ISSUE THREAD

**Issue by ajmeese7** [openemr/openemr#6030]
### Describe the problem

Currently the Patient ID generated when a patient is created is random, associating it back to a specific user via username requires some form of database so I know how to get that patient's info when they log in.

### Describe the solution you'd like

I would like to be able to specify the patient's username as their PID, that way an additional username to PID correlation is not required.

If I'm misunderstanding something please feel free to correct me! I could very well be wrong.

**Comment by sjpadgett** [comment]
You may be able to use the external id in demographics

![image](https://user-images.githubusercontent.com/1859985/208199102-efc80c31-642b-4092-a597-75d3c59526a7.png)


![OpenEMR](https://user-images.githubusercontent.com/1859985/208199507-3c0a99b9-14aa-4f00-8a80-5baa12fa5003.png)

**Comment by ajmeese7** [comment]
I saw that option and thought it was interesting, but I need to be able to get a patient by `{puuid}` according to the `/api/patient` PUT endpoint docs. Do you know if there's a way to use the external ID for that purpose instead?

**Comment by sjpadgett** [comment]
It'd have to be added as a query option to the end point. I haven't looked at the endpoits in awhile but it probably should be the primary id on patients as it is the external public MRN.
That'd be something to put in front of the rest of the team and would require a refactor or additional query arg added specific to MRN.

You could put up a PR and see where it goes.

**Comment by robertdown** [comment]
I don't believe PID is random, it's auto-incrementing. The account name is generated by first name + PID. 

PID is automatically created with a patient. The External ID field (internally also called "pubpid") is what you may be thinking of. I believe that field autopopulates with PID if left blank.

Is there a reason you'd want an account name to _not_ be generated with a unique value?

**Comment by sjpadgett** [comment]
@robertdown is correct the external id is/was meant to be the patient public MRN and can be used to associate a patient to an external reference of perhaps a specialist or EHR. CDA imports will use the patient UUID which is known to the sending end for later reference if needed.
Default is the assigned internal pid and is by no means random.
For some of our international users(recently brought up to me) they find having the public the same as the pid confusing and prefer a more human readable MRN

**Comment by robertdown** [comment]
I think a masked solution for PUBPID that builds a number based on some rules the business sets up is a worthwhile endeavor. May add that to my list of todo...

**Comment by sjpadgett** [comment]
I thought that is already available via globals setup!

**Comment by robertdown** [comment]
Well look at that....

**Comment by robertdown** [comment]
Actually, that's just a rules enforcement mask, it will **not** auto-generate a number for you.

**Comment by ajmeese7** [comment]
> Is there a reason you'd want an account name to not be generated with a unique value?

@robertdown It's not that I don't want a unique value, I simply want to *specify* the unique value. The patient username for my system is already unique, so I don't want to have to create a database just to correlate a username to a PID.
